lib/elixir/test/elixir/dynamic_supervisor_test.exs

```diff
@@ -356,6 +356,27 @@ defmodule DynamicSupervisorTest do
       assert {:error, :max_children} = DynamicSupervisor.start_child(pid, child)
     end
 
+    # regression test for #7369
+    test "restarting a child with extra_args does not crash the supervisor" do
```

Comment:
Could we rename this test as `restarting a child with extra_args successfully restarts child`. The supervisor was not crashing, rather it would repeatedly try to restart a child with the wrong arity until it reached max intensity. The current name is a little confusing without the context of the issue.

Reply:
Done.